plugins/terms/server/__init__.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import base64
+import datetime
+import hashlib
+
+from girder import events
+from girder.api import access
+from girder.api.describe import Description, autoDescribeRoute
+from girder.api.rest import boundHandler, RestException
+from girder.api.v1.collection import Collection
+from girder.constants import AccessType, TokenScope
+from girder.models.model_base import ModelImporter
+
+
+@access.user(scope=TokenScope.DATA_READ)
+@boundHandler
+@autoDescribeRoute(
+    Description('Accept a collection\'s Terms of Use for the current user.')
+    .modelParam('id', model='collection', level=AccessType.READ)
+    .param('termsHash', 'The SHA-256 hash of this collection\'s terms, encoded in Base64.')
+)
+def acceptCollectionTerms(self, collection, termsHash):
+    if not collection.get('terms'):
+        raise RestException('This collection currently has no terms.')
+
+    # termsHash should be encoded to a bytes object, but storing bytes into MongoDB behaves
+    # differently in Python 2 vs 3. Additionally, serializing a bytes to JSON behaves differently
+    # in Python 2 vs 3. So, just keep it as a unicode (or ordinary Python 2 str).
+    realTermsHash = base64.b64encode(
+        hashlib.sha256(collection['terms'].encode('utf-8')).digest()
+    ).decode('utf-8')
```

Comment:
Would it be ok to just use the hex digest instead, since most people are used to seeing that as the text form of checksums?